src/radical/pilot/agent/lm/spark.py

```diff
@@ -0,0 +1,285 @@
+
+__copyright__ = "Copyright 2016, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+
+import os
+import subprocess
+import urllib
+import sys
+import socket
+import random
+
+from .base import LaunchMethod
+
+
+
+# ==============================================================================
+#
+# The Launch Method Implementation for Running Spark applications
+#
+class Spark(LaunchMethod):
+
+        # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, logger):
+
+        LaunchMethod.__init__(self, cfg, logger)
+
+
+
+    # --------------------------------------------------------------------------
+    #
+    @classmethod
+    def lrms_config_hook(cls, name, cfg, lrms, logger):
+        """
+        FIXME: this config hook will inspect the LRMS nodelist and, if needed,
+               will start the SPARK cluster on node[0].
+        """
+        import radical.utils as ru
+
+        logger.info('Hook called by SPARK LRMS with the name %s'%lrms.name)
+
+        # If the LRMS used is not SPARK the namenode url is going to be
+        # the first node in the list and the port is the default one, else 
+        # it is the one that the SPARK LRMS returns
+        spark_home = None   
+        if lrms.name == 'SPARKLRMS':
+            logger.info("Found SPARK ")
```

Comment:
I could not find where `SPARLRMS` is ever set in the source code.  Could you clarify this, please?